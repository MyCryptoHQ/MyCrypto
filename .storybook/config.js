import React from 'react';
import { addDecorator } from '@storybook/react';
import { withA11y } from '@storybook/addon-a11y';
import { withThemesProvider } from 'storybook-addon-styled-component-theme';
import styled from 'styled-components';
import { light, dark } from '@mycrypto/ui';

import { GAU_COLORS } from 'v2/theme';

import 'sass/styles'

const LIGHT_THEME = {
  name: 'Light',
  ...light,
  GAU: {
    COLORS: GAU_COLORS
  }
};

const DARK_THEME = {
  name: 'Dark',
  ...dark,
  GAU: {
    COLORS: GAU_COLORS
  }
};

addDecorator(withThemesProvider([LIGHT_THEME, DARK_THEME]));
addDecorator(withA11y);
