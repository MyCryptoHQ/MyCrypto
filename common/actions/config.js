// @flow

/*** Change Language ***/
export type ChangeLanguageAction = {
  type: 'CONFIG_LANGUAGE_CHANGE',
  value: string
};

export function changeLanguage(sign: string): ChangeLanguageAction {
  return {
    type: 'CONFIG_LANGUAGE_CHANGE',
    value: sign
  };
}

/*** Change Node ***/
export type ChangeNodeAction = {
  type: 'CONFIG_NODE_CHANGE',
  // FIXME $keyof?
  value: string
};

export function changeNode(value: string): ChangeNodeAction {
  return {
    type: 'CONFIG_NODE_CHANGE',
    value
  };
}

/*** Change gas price ***/
export type ChangeGasPriceAction = {
  type: 'CONFIG_GAS_PRICE',
  value: number
};

export function changeGasPrice(value: number): ChangeGasPriceAction {
  return {
    type: 'CONFIG_GAS_PRICE',
    value
  };
}

/*** Union Type ***/
export type ConfigAction =
  | ChangeNodeAction
  | ChangeLanguageAction
  | ChangeGasPriceAction;
