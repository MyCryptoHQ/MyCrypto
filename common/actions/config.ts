/*** Change Language ***/
export interface ChangeLanguageAction {
  type: 'CONFIG_LANGUAGE_CHANGE';
  value: string;
}

export function changeLanguage(sign: string): ChangeLanguageAction {
  return {
    type: 'CONFIG_LANGUAGE_CHANGE',
    value: sign
  };
}

/*** Change Node ***/
export interface ChangeNodeAction {
  type: 'CONFIG_NODE_CHANGE';
  // FIXME $keyof?
  value: string;
}

export function changeNode(value: string): ChangeNodeAction {
  return {
    type: 'CONFIG_NODE_CHANGE',
    value
  };
}

/*** Change gas price ***/
export interface ChangeGasPriceAction {
  type: 'CONFIG_GAS_PRICE';
  value: number;
}

export function changeGasPrice(value: number): ChangeGasPriceAction {
  return {
    type: 'CONFIG_GAS_PRICE',
    value
  };
}

/*** Union Type ***/
export type ConfigAction =
  | ChangeNodeAction
  | ChangeLanguageAction
  | ChangeGasPriceAction;
