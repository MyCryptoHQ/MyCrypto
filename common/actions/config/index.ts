export * from './actionCreators';
export * from './actionTypes';
