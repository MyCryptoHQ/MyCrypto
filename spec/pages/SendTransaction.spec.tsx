import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import SendTransaction from 'containers/Tabs/SendTransaction';
import shallowWithStore from '../utils/shallowWithStore';
import { createMockStore } from 'redux-test-utils';
import { NODES } from 'config/data';
import { RouteComponentProps } from 'react-router';
import { createMockRouteComponentProps } from '../utils/mockRouteComponentProps';

Enzyme.configure({ adapter: new Adapter() });

it('render snapshot', () => {
  const testNode = 'rop_mew';
  const testStateConfig = {
    languageSelection: 'en',
    nodeSelection: testNode,
    node: NODES[testNode],
    gasPriceGwei: 21,
    offline: false,
    forceOffline: false
  };
  const testState = {
    wallet: {},
    balance: {},
    tokenBalances: {},
    node: {},
    nodeLib: {},
    network: {},
    tokens: [],
    gasPrice: {},
    transactions: {},
    offline: {},
    forceOffline: {},
    config: testStateConfig,
    customTokens: []
  };
  const routeProps: RouteComponentProps<any> = createMockRouteComponentProps({
    match: { path: '/account', url: '/account', isExact: true, params: {} },
    location: { pathname: '/account', search: '', hash: '', key: 'e08jz7' },
    history: {
      length: 2,
      action: 'PUSH',
      location: { pathname: '/account', search: '', hash: '', key: 'e08jz7', state: {} }
    }
  });

  const store = createMockStore(testState);
  const component = shallowWithStore(<SendTransaction {...routeProps} />, store);

  expect(component).toMatchSnapshot();
});
