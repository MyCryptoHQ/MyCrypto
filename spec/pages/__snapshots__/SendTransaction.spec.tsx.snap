// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render snapshot 1`] = `
<SendTransaction
  broadcastTx={[Function]}
  forceOffline={false}
  gasPrice={"4e3b29200"}
  location={
    Object {
      "search": "?to=73640ebefe93e4d0d6e9030ee9c1866ad1f3b9f1feeb403e978c4952d8369b39",
    }
  }
  network={
    Object {
      "blockExplorer": Object {
        "address": [Function],
        "name": "https://ropsten.etherscan.io",
        "tx": [Function],
      },
      "chainId": 3,
      "color": "#adc101",
      "contracts": Array [
        Object {
          "abi": "[{\\"constant\\":true,\\"inputs\\":[],\\"name\\":\\"ens\\",\\"outputs\\":[{\\"name\\":\\"\\",\\"type\\":\\"address\\"}],\\"payable\\":false,\\"type\\":\\"function\\"},{\\"constant\\":true,\\"inputs\\":[{\\"name\\":\\"\\",\\"type\\":\\"bytes32\\"}],\\"name\\":\\"expiryTimes\\",\\"outputs\\":[{\\"name\\":\\"\\",\\"type\\":\\"uint256\\"}],\\"payable\\":false,\\"type\\":\\"function\\"},{\\"constant\\":false,\\"inputs\\":[{\\"name\\":\\"subnode\\",\\"type\\":\\"bytes32\\"},{\\"name\\":\\"owner\\",\\"type\\":\\"address\\"}],\\"name\\":\\"register\\",\\"outputs\\":[],\\"payable\\":false,\\"type\\":\\"function\\"},{\\"constant\\":true,\\"inputs\\":[],\\"name\\":\\"rootNode\\",\\"outputs\\":[{\\"name\\":\\"\\",\\"type\\":\\"bytes32\\"}],\\"payable\\":false,\\"type\\":\\"function\\"},{\\"inputs\\":[{\\"name\\":\\"ensAddr\\",\\"type\\":\\"address\\"},{\\"name\\":\\"node\\",\\"type\\":\\"bytes32\\"}],\\"type\\":\\"constructor\\"}]",
          "address": "0x21397c1a1f4acd9132fe36df011610564b87e24b",
          "name": "??? ENS - Eth Registrar (Auction) ??? ",
        },
        Object {
          "abi": "[{\\"constant\\":false,\\"inputs\\":[{\\"name\\":\\"_hash\\",\\"type\\":\\"bytes32\\"}],\\"name\\":\\"releaseDeed\\",\\"outputs\\":[],\\"payable\\":false,\\"type\\":\\"function\\"},{\\"constant\\":true,\\"inputs\\":[{\\"name\\":\\"_hash\\",\\"type\\":\\"bytes32\\"}],\\"name\\":\\"getAllowedTime\\",\\"outputs\\":[{\\"name\\":\\"timestamp\\",\\"type\\":\\"uint256\\"}],\\"payable\\":false,\\"type\\":\\"function\\"},{\\"constant\\":false,\\"inputs\\":[{\\"name\\":\\"unhashedName\\",\\"type\\":\\"string\\"}],\\"name\\":\\"invalidateName\\",\\"outputs\\":[],\\"payable\\":false,\\"type\\":\\"function\\"},{\\"constant\\":true,\\"inputs\\":[{\\"name\\":\\"hash\\",\\"type\\":\\"bytes32\\"},{\\"name\\":\\"owner\\",\\"type\\":\\"address\\"},{\\"name\\":\\"value\\",\\"type\\":\\"uint256\\"},{\\"name\\":\\"salt\\",\\"type\\":\\"bytes32\\"}],\\"name\\":\\"shaBid\\",\\"outputs\\":[{\\"name\\":\\"sealedBid\\",\\"type\\":\\"bytes32\\"}],\\"payable\\":false,\\"type\\":\\"function\\"},{\\"constant\\":false,\\"inputs\\":[{\\"name\\":\\"bidder\\",\\"type\\":\\"address\\"},{\\"name\\":\\"seal\\",\\"type\\":\\"bytes32\\"}],\\"name\\":\\"cancelBid\\",\\"outputs\\":[],\\"payable\\":false,\\"type\\":\\"function\\"},{\\"constant\\":true,\\"inputs\\":[{\\"name\\":\\"_hash\\",\\"type\\":\\"bytes32\\"}],\\"name\\":\\"entries\\",\\"outputs\\":[{\\"name\\":\\"\\",\\"type\\":\\"uint8\\"},{\\"name\\":\\"\\",\\"type\\":\\"address\\"},{\\"name\\":\\"\\",\\"type\\":\\"uint256\\"},{\\"name\\":\\"\\",\\"type\\":\\"uint256\\"},{\\"name\\":\\"\\",\\"type\\":\\"uint256\\"}],\\"payable\\":false,\\"type\\":\\"function\\"},{\\"constant\\":true,\\"inputs\\":[],\\"name\\":\\"ens\\",\\"outputs\\":[{\\"name\\":\\"\\",\\"type\\":\\"address\\"}],\\"payable\\":false,\\"type\\":\\"function\\"},{\\"constant\\":false,\\"inputs\\":[{\\"name\\":\\"_hash\\",\\"type\\":\\"bytes32\\"},{\\"name\\":\\"_value\\",\\"type\\":\\"uint256\\"},{\\"name\\":\\"_salt\\",\\"type\\":\\"bytes32\\"}],\\"name\\":\\"unsealBid\\",\\"outputs\\":[],\\"payable\\":false,\\"type\\":\\"function\\"},{\\"constant\\":false,\\"inputs\\":[{\\"name\\":\\"_hash\\",\\"type\\":\\"bytes32\\"}],\\"name\\":\\"transferRegistrars\\",\\"outputs\\":[],\\"payable\\":false,\\"type\\":\\"function\\"},{\\"constant\\":true,\\"inputs\\":[{\\"name\\":\\"\\",\\"type\\":\\"address\\"},{\\"name\\":\\"\\",\\"type\\":\\"bytes32\\"}],\\"name\\":\\"sealedBids\\",\\"outputs\\":[{\\"name\\":\\"\\",\\"type\\":\\"address\\"}],\\"payable\\":false,\\"type\\":\\"function\\"},{\\"constant\\":true,\\"inputs\\":[{\\"name\\":\\"_hash\\",\\"type\\":\\"bytes32\\"}],\\"name\\":\\"state\\",\\"outputs\\":[{\\"name\\":\\"\\",\\"type\\":\\"uint8\\"}],\\"payable\\":false,\\"type\\":\\"function\\"},{\\"constant\\":false,\\"inputs\\":[{\\"name\\":\\"_hash\\",\\"type\\":\\"bytes32\\"},{\\"name\\":\\"newOwner\\",\\"type\\":\\"address\\"}],\\"name\\":\\"transfer\\",\\"outputs\\":[],\\"payable\\":false,\\"type\\":\\"function\\"},{\\"constant\\":true,\\"inputs\\":[{\\"name\\":\\"_hash\\",\\"type\\":\\"bytes32\\"},{\\"name\\":\\"_timestamp\\",\\"type\\":\\"uint256\\"}],\\"name\\":\\"isAllowed\\",\\"outputs\\":[{\\"name\\":\\"allowed\\",\\"type\\":\\"bool\\"}],\\"payable\\":false,\\"type\\":\\"function\\"},{\\"constant\\":false,\\"inputs\\":[{\\"name\\":\\"_hash\\",\\"type\\":\\"bytes32\\"}],\\"name\\":\\"finalizeAuction\\",\\"outputs\\":[],\\"payable\\":false,\\"type\\":\\"function\\"},{\\"constant\\":true,\\"inputs\\":[],\\"name\\":\\"registryStarted\\",\\"outputs\\":[{\\"name\\":\\"\\",\\"type\\":\\"uint256\\"}],\\"payable\\":false,\\"type\\":\\"function\\"},{\\"constant\\":true,\\"inputs\\":[],\\"name\\":\\"launchLength\\",\\"outputs\\":[{\\"name\\":\\"\\",\\"type\\":\\"uint32\\"}],\\"payable\\":false,\\"type\\":\\"function\\"},{\\"constant\\":false,\\"inputs\\":[{\\"name\\":\\"sealedBid\\",\\"type\\":\\"bytes32\\"}],\\"name\\":\\"newBid\\",\\"outputs\\":[],\\"payable\\":true,\\"type\\":\\"function\\"},{\\"constant\\":false,\\"inputs\\":[{\\"name\\":\\"labels\\",\\"type\\":\\"bytes32[]\\"}],\\"name\\":\\"eraseNode\\",\\"outputs\\":[],\\"payable\\":false,\\"type\\":\\"function\\"},{\\"constant\\":false,\\"inputs\\":[{\\"name\\":\\"_hashes\\",\\"type\\":\\"bytes32[]\\"}],\\"name\\":\\"startAuctions\\",\\"outputs\\":[],\\"payable\\":false,\\"type\\":\\"function\\"},{\\"constant\\":false,\\"inputs\\":[{\\"name\\":\\"hash\\",\\"type\\":\\"bytes32\\"},{\\"name\\":\\"deed\\",\\"type\\":\\"address\\"},{\\"name\\":\\"registrationDate\\",\\"type\\":\\"uint256\\"}],\\"name\\":\\"acceptRegistrarTransfer\\",\\"outputs\\":[],\\"payable\\":false,\\"type\\":\\"function\\"},{\\"constant\\":false,\\"inputs\\":[{\\"name\\":\\"_hash\\",\\"type\\":\\"bytes32\\"}],\\"name\\":\\"startAuction\\",\\"outputs\\":[],\\"payable\\":false,\\"type\\":\\"function\\"},{\\"constant\\":true,\\"inputs\\":[],\\"name\\":\\"rootNode\\",\\"outputs\\":[{\\"name\\":\\"\\",\\"type\\":\\"bytes32\\"}],\\"payable\\":false,\\"type\\":\\"function\\"},{\\"constant\\":false,\\"inputs\\":[{\\"name\\":\\"hashes\\",\\"type\\":\\"bytes32[]\\"},{\\"name\\":\\"sealedBid\\",\\"type\\":\\"bytes32\\"}],\\"name\\":\\"startAuctionsAndBid\\",\\"outputs\\":[],\\"payable\\":true,\\"type\\":\\"function\\"},{\\"inputs\\":[{\\"name\\":\\"_ens\\",\\"type\\":\\"address\\"},{\\"name\\":\\"_rootNode\\",\\"type\\":\\"bytes32\\"},{\\"name\\":\\"_startDate\\",\\"type\\":\\"uint256\\"}],\\"payable\\":false,\\"type\\":\\"constructor\\"},{\\"anonymous\\":false,\\"inputs\\":[{\\"indexed\\":true,\\"name\\":\\"hash\\",\\"type\\":\\"bytes32\\"},{\\"indexed\\":false,\\"name\\":\\"registrationDate\\",\\"type\\":\\"uint256\\"}],\\"name\\":\\"AuctionStarted\\",\\"type\\":\\"event\\"},{\\"anonymous\\":false,\\"inputs\\":[{\\"indexed\\":true,\\"name\\":\\"hash\\",\\"type\\":\\"bytes32\\"},{\\"indexed\\":true,\\"name\\":\\"bidder\\",\\"type\\":\\"address\\"},{\\"indexed\\":false,\\"name\\":\\"deposit\\",\\"type\\":\\"uint256\\"}],\\"name\\":\\"NewBid\\",\\"type\\":\\"event\\"},{\\"anonymous\\":false,\\"inputs\\":[{\\"indexed\\":true,\\"name\\":\\"hash\\",\\"type\\":\\"bytes32\\"},{\\"indexed\\":true,\\"name\\":\\"owner\\",\\"type\\":\\"address\\"},{\\"indexed\\":false,\\"name\\":\\"value\\",\\"type\\":\\"uint256\\"},{\\"indexed\\":false,\\"name\\":\\"status\\",\\"type\\":\\"uint8\\"}],\\"name\\":\\"BidRevealed\\",\\"type\\":\\"event\\"},{\\"anonymous\\":false,\\"inputs\\":[{\\"indexed\\":true,\\"name\\":\\"hash\\",\\"type\\":\\"bytes32\\"},{\\"indexed\\":true,\\"name\\":\\"owner\\",\\"type\\":\\"address\\"},{\\"indexed\\":false,\\"name\\":\\"value\\",\\"type\\":\\"uint256\\"},{\\"indexed\\":false,\\"name\\":\\"registrationDate\\",\\"type\\":\\"uint256\\"}],\\"name\\":\\"HashRegistered\\",\\"type\\":\\"event\\"},{\\"anonymous\\":false,\\"inputs\\":[{\\"indexed\\":true,\\"name\\":\\"hash\\",\\"type\\":\\"bytes32\\"},{\\"indexed\\":false,\\"name\\":\\"value\\",\\"type\\":\\"uint256\\"}],\\"name\\":\\"HashReleased\\",\\"type\\":\\"event\\"},{\\"anonymous\\":false,\\"inputs\\":[{\\"indexed\\":true,\\"name\\":\\"hash\\",\\"type\\":\\"bytes32\\"},{\\"indexed\\":true,\\"name\\":\\"name\\",\\"type\\":\\"string\\"},{\\"indexed\\":false,\\"name\\":\\"value\\",\\"type\\":\\"uint256\\"},{\\"indexed\\":false,\\"name\\":\\"registrationDate\\",\\"type\\":\\"uint256\\"}],\\"name\\":\\"HashInvalidated\\",\\"type\\":\\"event\\"}]",
          "address": "0xc19fd9004b5c9789391679de6d766b981db94610",
          "name": "??? ENS - Eth Registrar (Auction) ??? ",
        },
        Object {
          "abi": "[{\\"constant\\":true,\\"inputs\\":[{\\"name\\":\\"node\\",\\"type\\":\\"bytes32\\"}],\\"name\\":\\"resolver\\",\\"outputs\\":[{\\"name\\":\\"\\",\\"type\\":\\"address\\"}],\\"payable\\":false,\\"type\\":\\"function\\"},{\\"constant\\":true,\\"inputs\\":[{\\"name\\":\\"node\\",\\"type\\":\\"bytes32\\"}],\\"name\\":\\"owner\\",\\"outputs\\":[{\\"name\\":\\"\\",\\"type\\":\\"address\\"}],\\"payable\\":false,\\"type\\":\\"function\\"},{\\"constant\\":false,\\"inputs\\":[{\\"name\\":\\"node\\",\\"type\\":\\"bytes32\\"},{\\"name\\":\\"label\\",\\"type\\":\\"bytes32\\"},{\\"name\\":\\"owner\\",\\"type\\":\\"address\\"}],\\"name\\":\\"setSubnodeOwner\\",\\"outputs\\":[],\\"payable\\":false,\\"type\\":\\"function\\"},{\\"constant\\":false,\\"inputs\\":[{\\"name\\":\\"node\\",\\"type\\":\\"bytes32\\"},{\\"name\\":\\"ttl\\",\\"type\\":\\"uint64\\"}],\\"name\\":\\"setTTL\\",\\"outputs\\":[],\\"payable\\":false,\\"type\\":\\"function\\"},{\\"constant\\":true,\\"inputs\\":[{\\"name\\":\\"node\\",\\"type\\":\\"bytes32\\"}],\\"name\\":\\"ttl\\",\\"outputs\\":[{\\"name\\":\\"\\",\\"type\\":\\"uint64\\"}],\\"payable\\":false,\\"type\\":\\"function\\"},{\\"constant\\":false,\\"inputs\\":[{\\"name\\":\\"node\\",\\"type\\":\\"bytes32\\"},{\\"name\\":\\"resolver\\",\\"type\\":\\"address\\"}],\\"name\\":\\"setResolver\\",\\"outputs\\":[],\\"payable\\":false,\\"type\\":\\"function\\"},{\\"constant\\":false,\\"inputs\\":[{\\"name\\":\\"node\\",\\"type\\":\\"bytes32\\"},{\\"name\\":\\"owner\\",\\"type\\":\\"address\\"}],\\"name\\":\\"setOwner\\",\\"outputs\\":[],\\"payable\\":false,\\"type\\":\\"function\\"},{\\"anonymous\\":false,\\"inputs\\":[{\\"indexed\\":true,\\"name\\":\\"node\\",\\"type\\":\\"bytes32\\"},{\\"indexed\\":false,\\"name\\":\\"owner\\",\\"type\\":\\"address\\"}],\\"name\\":\\"Transfer\\",\\"type\\":\\"event\\"},{\\"anonymous\\":false,\\"inputs\\":[{\\"indexed\\":true,\\"name\\":\\"node\\",\\"type\\":\\"bytes32\\"},{\\"indexed\\":true,\\"name\\":\\"label\\",\\"type\\":\\"bytes32\\"},{\\"indexed\\":false,\\"name\\":\\"owner\\",\\"type\\":\\"address\\"}],\\"name\\":\\"NewOwner\\",\\"type\\":\\"event\\"},{\\"anonymous\\":false,\\"inputs\\":[{\\"indexed\\":true,\\"name\\":\\"node\\",\\"type\\":\\"bytes32\\"},{\\"indexed\\":false,\\"name\\":\\"resolver\\",\\"type\\":\\"address\\"}],\\"name\\":\\"NewResolver\\",\\"type\\":\\"event\\"},{\\"anonymous\\":false,\\"inputs\\":[{\\"indexed\\":true,\\"name\\":\\"node\\",\\"type\\":\\"bytes32\\"},{\\"indexed\\":false,\\"name\\":\\"ttl\\",\\"type\\":\\"uint64\\"}],\\"name\\":\\"NewTTL\\",\\"type\\":\\"event\\"}]",
          "address": "0x112234455c3a32fd11230c42e7bccd4a84e02010",
          "name": "ENS - Registry",
        },
        Object {
          "abi": "[{\\"constant\\":true,\\"inputs\\":[{\\"name\\":\\"interfaceID\\",\\"type\\":\\"bytes4\\"}],\\"name\\":\\"supportsInterface\\",\\"outputs\\":[{\\"name\\":\\"\\",\\"type\\":\\"bool\\"}],\\"payable\\":false,\\"type\\":\\"function\\"},{\\"constant\\":true,\\"inputs\\":[{\\"name\\":\\"node\\",\\"type\\":\\"bytes32\\"}],\\"name\\":\\"addr\\",\\"outputs\\":[{\\"name\\":\\"ret\\",\\"type\\":\\"address\\"}],\\"payable\\":false,\\"type\\":\\"function\\"},{\\"constant\\":true,\\"inputs\\":[{\\"name\\":\\"node\\",\\"type\\":\\"bytes32\\"},{\\"name\\":\\"kind\\",\\"type\\":\\"bytes32\\"}],\\"name\\":\\"has\\",\\"outputs\\":[{\\"name\\":\\"\\",\\"type\\":\\"bool\\"}],\\"payable\\":false,\\"type\\":\\"function\\"},{\\"constant\\":false,\\"inputs\\":[{\\"name\\":\\"node\\",\\"type\\":\\"bytes32\\"},{\\"name\\":\\"addr\\",\\"type\\":\\"address\\"}],\\"name\\":\\"setAddr\\",\\"outputs\\":[],\\"payable\\":false,\\"type\\":\\"function\\"},{\\"constant\\":true,\\"inputs\\":[{\\"name\\":\\"node\\",\\"type\\":\\"bytes32\\"}],\\"name\\":\\"content\\",\\"outputs\\":[{\\"name\\":\\"ret\\",\\"type\\":\\"bytes32\\"}],\\"payable\\":false,\\"type\\":\\"function\\"},{\\"constant\\":false,\\"inputs\\":[{\\"name\\":\\"node\\",\\"type\\":\\"bytes32\\"},{\\"name\\":\\"hash\\",\\"type\\":\\"bytes32\\"}],\\"name\\":\\"setContent\\",\\"outputs\\":[],\\"payable\\":false,\\"type\\":\\"function\\"},{\\"inputs\\":[{\\"name\\":\\"ensAddr\\",\\"type\\":\\"address\\"}],\\"type\\":\\"constructor\\"},{\\"payable\\":false,\\"type\\":\\"fallback\\"}]",
          "address": "0x4c641fb9bad9b60ef180c31f56051ce826d21a9a",
          "name": "ENS - Public Resolver",
        },
        Object {
          "abi": "[{\\"constant\\":false,\\"inputs\\":[{\\"name\\":\\"_owner\\",\\"type\\":\\"address\\"}],\\"name\\":\\"removeOwner\\",\\"outputs\\":[],\\"type\\":\\"function\\"},{\\"constant\\":false,\\"inputs\\":[{\\"name\\":\\"_addr\\",\\"type\\":\\"address\\"}],\\"name\\":\\"isOwner\\",\\"outputs\\":[{\\"name\\":\\"\\",\\"type\\":\\"bool\\"}],\\"type\\":\\"function\\"},{\\"constant\\":true,\\"inputs\\":[],\\"name\\":\\"m_numOwners\\",\\"outputs\\":[{\\"name\\":\\"\\",\\"type\\":\\"uint256\\"}],\\"type\\":\\"function\\"},{\\"constant\\":true,\\"inputs\\":[],\\"name\\":\\"m_lastDay\\",\\"outputs\\":[{\\"name\\":\\"\\",\\"type\\":\\"uint256\\"}],\\"type\\":\\"function\\"},{\\"constant\\":true,\\"inputs\\":[],\\"name\\":\\"version\\",\\"outputs\\":[{\\"name\\":\\"\\",\\"type\\":\\"uint256\\"}],\\"type\\":\\"function\\"},{\\"constant\\":false,\\"inputs\\":[],\\"name\\":\\"resetSpentToday\\",\\"outputs\\":[],\\"type\\":\\"function\\"},{\\"constant\\":true,\\"inputs\\":[],\\"name\\":\\"m_spentToday\\",\\"outputs\\":[{\\"name\\":\\"\\",\\"type\\":\\"uint256\\"}],\\"type\\":\\"function\\"},{\\"constant\\":false,\\"inputs\\":[{\\"name\\":\\"_owner\\",\\"type\\":\\"address\\"}],\\"name\\":\\"addOwner\\",\\"outputs\\":[],\\"type\\":\\"function\\"},{\\"constant\\":true,\\"inputs\\":[],\\"name\\":\\"m_required\\",\\"outputs\\":[{\\"name\\":\\"\\",\\"type\\":\\"uint256\\"}],\\"type\\":\\"function\\"},{\\"constant\\":false,\\"inputs\\":[{\\"name\\":\\"_h\\",\\"type\\":\\"bytes32\\"}],\\"name\\":\\"confirm\\",\\"outputs\\":[{\\"name\\":\\"\\",\\"type\\":\\"bool\\"}],\\"type\\":\\"function\\"},{\\"constant\\":false,\\"inputs\\":[{\\"name\\":\\"_newLimit\\",\\"type\\":\\"uint256\\"}],\\"name\\":\\"setDailyLimit\\",\\"outputs\\":[],\\"type\\":\\"function\\"},{\\"constant\\":false,\\"inputs\\":[{\\"name\\":\\"_to\\",\\"type\\":\\"address\\"},{\\"name\\":\\"_value\\",\\"type\\":\\"uint256\\"},{\\"name\\":\\"_data\\",\\"type\\":\\"bytes\\"}],\\"name\\":\\"execute\\",\\"outputs\\":[{\\"name\\":\\"_r\\",\\"type\\":\\"bytes32\\"}],\\"type\\":\\"function\\"},{\\"constant\\":false,\\"inputs\\":[{\\"name\\":\\"_operation\\",\\"type\\":\\"bytes32\\"}],\\"name\\":\\"revoke\\",\\"outputs\\":[],\\"type\\":\\"function\\"},{\\"constant\\":false,\\"inputs\\":[{\\"name\\":\\"_newRequired\\",\\"type\\":\\"uint256\\"}],\\"name\\":\\"changeRequirement\\",\\"outputs\\":[],\\"type\\":\\"function\\"},{\\"constant\\":true,\\"inputs\\":[{\\"name\\":\\"_operation\\",\\"type\\":\\"bytes32\\"},{\\"name\\":\\"_owner\\",\\"type\\":\\"address\\"}],\\"name\\":\\"hasConfirmed\\",\\"outputs\\":[{\\"name\\":\\"\\",\\"type\\":\\"bool\\"}],\\"type\\":\\"function\\"},{\\"constant\\":false,\\"inputs\\":[{\\"name\\":\\"_to\\",\\"type\\":\\"address\\"}],\\"name\\":\\"kill\\",\\"outputs\\":[],\\"type\\":\\"function\\"},{\\"constant\\":false,\\"inputs\\":[{\\"name\\":\\"_from\\",\\"type\\":\\"address\\"},{\\"name\\":\\"_to\\",\\"type\\":\\"address\\"}],\\"name\\":\\"changeOwner\\",\\"outputs\\":[],\\"type\\":\\"function\\"},{\\"constant\\":true,\\"inputs\\":[],\\"name\\":\\"m_dailyLimit\\",\\"outputs\\":[{\\"name\\":\\"\\",\\"type\\":\\"uint256\\"}],\\"type\\":\\"function\\"},{\\"inputs\\":[{\\"name\\":\\"_owners\\",\\"type\\":\\"address[]\\"},{\\"name\\":\\"_required\\",\\"type\\":\\"uint256\\"},{\\"name\\":\\"_daylimit\\",\\"type\\":\\"uint256\\"}],\\"type\\":\\"constructor\\"},{\\"anonymous\\":false,\\"inputs\\":[{\\"indexed\\":false,\\"name\\":\\"owner\\",\\"type\\":\\"address\\"},{\\"indexed\\":false,\\"name\\":\\"operation\\",\\"type\\":\\"bytes32\\"}],\\"name\\":\\"Confirmation\\",\\"type\\":\\"event\\"},{\\"anonymous\\":false,\\"inputs\\":[{\\"indexed\\":false,\\"name\\":\\"owner\\",\\"type\\":\\"address\\"},{\\"indexed\\":false,\\"name\\":\\"operation\\",\\"type\\":\\"bytes32\\"}],\\"name\\":\\"Revoke\\",\\"type\\":\\"event\\"},{\\"anonymous\\":false,\\"inputs\\":[{\\"indexed\\":false,\\"name\\":\\"oldOwner\\",\\"type\\":\\"address\\"},{\\"indexed\\":false,\\"name\\":\\"newOwner\\",\\"type\\":\\"address\\"}],\\"name\\":\\"OwnerChanged\\",\\"type\\":\\"event\\"},{\\"anonymous\\":false,\\"inputs\\":[{\\"indexed\\":false,\\"name\\":\\"newOwner\\",\\"type\\":\\"address\\"}],\\"name\\":\\"OwnerAdded\\",\\"type\\":\\"event\\"},{\\"anonymous\\":false,\\"inputs\\":[{\\"indexed\\":false,\\"name\\":\\"oldOwner\\",\\"type\\":\\"address\\"}],\\"name\\":\\"OwnerRemoved\\",\\"type\\":\\"event\\"},{\\"anonymous\\":false,\\"inputs\\":[{\\"indexed\\":false,\\"name\\":\\"newRequirement\\",\\"type\\":\\"uint256\\"}],\\"name\\":\\"RequirementChanged\\",\\"type\\":\\"event\\"},{\\"anonymous\\":false,\\"inputs\\":[{\\"indexed\\":false,\\"name\\":\\"from\\",\\"type\\":\\"address\\"},{\\"indexed\\":false,\\"name\\":\\"value\\",\\"type\\":\\"uint256\\"}],\\"name\\":\\"Deposit\\",\\"type\\":\\"event\\"},{\\"anonymous\\":false,\\"inputs\\":[{\\"indexed\\":false,\\"name\\":\\"owner\\",\\"type\\":\\"address\\"},{\\"indexed\\":false,\\"name\\":\\"value\\",\\"type\\":\\"uint256\\"},{\\"indexed\\":false,\\"name\\":\\"to\\",\\"type\\":\\"address\\"},{\\"indexed\\":false,\\"name\\":\\"data\\",\\"type\\":\\"bytes\\"}],\\"name\\":\\"SingleTransact\\",\\"type\\":\\"event\\"},{\\"anonymous\\":false,\\"inputs\\":[{\\"indexed\\":false,\\"name\\":\\"owner\\",\\"type\\":\\"address\\"},{\\"indexed\\":false,\\"name\\":\\"operation\\",\\"type\\":\\"bytes32\\"},{\\"indexed\\":false,\\"name\\":\\"value\\",\\"type\\":\\"uint256\\"},{\\"indexed\\":false,\\"name\\":\\"to\\",\\"type\\":\\"address\\"},{\\"indexed\\":false,\\"name\\":\\"data\\",\\"type\\":\\"bytes\\"}],\\"name\\":\\"MultiTransact\\",\\"type\\":\\"event\\"},{\\"anonymous\\":false,\\"inputs\\":[{\\"indexed\\":false,\\"name\\":\\"operation\\",\\"type\\":\\"bytes32\\"},{\\"indexed\\":false,\\"name\\":\\"initiator\\",\\"type\\":\\"address\\"},{\\"indexed\\":false,\\"name\\":\\"value\\",\\"type\\":\\"uint256\\"},{\\"indexed\\":false,\\"name\\":\\"to\\",\\"type\\":\\"address\\"},{\\"indexed\\":false,\\"name\\":\\"data\\",\\"type\\":\\"bytes\\"}],\\"name\\":\\"ConfirmationNeeded\\",\\"type\\":\\"event\\"}]",
          "address": "0x0000000000000000000000000000000000000000",
          "name": "Mist's Multisig Contract",
        },
      ],
      "name": "Ropsten",
      "tokens": Array [],
      "unit": "ETH",
    }
  }
  nodeLib={
    RpcNode {
      "client": RPCClient {
        "batch": [Function],
        "call": [Function],
        "createHeaders": [Function],
        "decorateRequest": [Function],
        "endpoint": "https://api.myetherapi.com/rop",
        "headers": Object {},
      },
      "requests": RPCRequests {},
    }
  }
  offline={false}
  pollOfflineStatus={[Function]}
  resetWallet={[Function]}
  showNotification={[Function]}
  tokenBalances={Array []}
  tokens={Array []}
/>
`;
